Mathlib/CategoryTheory/Limits/Presheaf.lean

```diff
@@ -444,111 +539,120 @@ noncomputable def natTrans : F.op.lan ⟶ G where
+instance : F.op.lan.IsLeftKanExtension (compULiftYonedaIsoULiftYonedaCompLan.{w} F).hom :=
   ⟨⟨Limits.IsInitial.ofUnique _⟩⟩
 
 end
 
-section
+/-- For a presheaf `P`, consider the forgetful functor from the category of representable
+    presheaves over `P` to the category of presheaves. There is a tautological cocone over this
+    functor whose leg for a natural transformation `V ⟶ P` with `V` representable is just that
+    natural transformation. (In this version, we allow the presheaf `P` to have values in
+    a larger universe.) -/
+@[simps]
+def tautologicalCocone' (P : Cᵒᵖ ⥤ Type max w v₁) :
+    Cocone (CostructuredArrow.proj uliftYoneda.{w} P ⋙ uliftYoneda.{w}) where
+  pt := P
+  ι := { app X := X.hom }
+
+/-- The tautological cocone with point `P` is a colimit cocone, exhibiting `P` as a colimit of
+    representables. (In this version, we allow the presheaf `P` to have values in
+    a larger universe.)
+
+    Proposition 2.6.3(i) in [Kashiwara2006] -/
+noncomputable def isColimitTautologicalCocone' (P : Cᵒᵖ ⥤ Type max w v₁) :
+    IsColimit (tautologicalCocone'.{w} P) :=
+  (IsColimit.whiskerEquivalenceEquiv
+    (CategoryOfElements.costructuredArrowULiftYonedaEquivalence.{w} P)).2
+      (colimitOfRepresentable.{w} P)
 
-variable {C : Type u₁} [Category.{v₁} C] (P : Cᵒᵖ ⥤ Type v₁)
 
 /-- For a presheaf `P`, consider the forgetful functor from the category of representable
     presheaves over `P` to the category of presheaves. There is a tautological cocone over this
     functor whose leg for a natural transformation `V ⟶ P` with `V` representable is just that
     natural transformation. -/
 @[simps]
-def tautologicalCocone : Cocone (CostructuredArrow.proj yoneda P ⋙ yoneda) where
+def tautologicalCocone (P : Cᵒᵖ ⥤ Type v₁) :
+    Cocone (CostructuredArrow.proj yoneda P ⋙ yoneda) where
   pt := P
-  ι := { app := fun X => X.hom }
+  ι := { app X := X.hom }
 
 /-- The tautological cocone with point `P` is a colimit cocone, exhibiting `P` as a colimit of
     representables.
 
     Proposition 2.6.3(i) in [Kashiwara2006] -/
-def isColimitTautologicalCocone : IsColimit (tautologicalCocone P) where
-  desc := fun s => by
-    refine ⟨fun X t => yonedaEquiv (s.ι.app (CostructuredArrow.mk (yonedaEquiv.symm t))), ?_⟩
-    intros X Y f
-    ext t
-    dsimp
-    rw [yonedaEquiv_naturality', yonedaEquiv_symm_map]
-    simpa using (s.ι.naturality
-      (CostructuredArrow.homMk' (CostructuredArrow.mk (yonedaEquiv.symm t)) f.unop)).symm
-  fac := by
-    intro s t
-    dsimp
-    apply yonedaEquiv.injective
-    rw [yonedaEquiv_comp]
-    dsimp only
-    rw [Equiv.symm_apply_apply]
-    rfl
-  uniq := by
-    intro s j h
-    ext V x
-    obtain ⟨t, rfl⟩ := yonedaEquiv.surjective x
-    dsimp
-    rw [Equiv.symm_apply_apply, ← yonedaEquiv_comp]
-    exact congr_arg _ (h (CostructuredArrow.mk t))
+noncomputable def isColimitTautologicalCocone (P : Cᵒᵖ ⥤ Type v₁) :
```

Comment:
```suggestion
def isColimitTautologicalCocone (P : Cᵒᵖ ⥤ Type v₁) :
```
This `noncomputable`, along with a few others, can ultimately be removed! :) See [eaa3d14](https://github.com/leanprover-community/mathlib4/pull/25780/commits/eaa3d142ece84ae4e49a7eaa3d07a014b7ec1334) (which is now based off of this PR).